app/javascript/components/ansible-credentials-form/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import MiqFormRenderer from '@@ddf';
+import createSchema from './ansible-credential-form.schema';
+import { initNotifications } from '../../miq-redux/actions/notifications-actions';
+
+const AnsibleCredentialsForm = ({ recordId }) => {
+  console.log(recordId);
+  const [{ fields, initialValues, loading }, setState] = useState({ fields: [], loading: !!recordId });
+  const promise = useMemo(() => API.options('/api/authentications'), []);
+
+  //make it pretty
+  const loadSchema = value => ({ data: { credential_types: { embedded_ansible_credential_types } } }) =>
+    setState(state => ({ ...state, fields: embedded_ansible_credential_types[value].attributes }));
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/authentications/${recordId}`).then(({ name, type, ...data }) => {
+        console.log(type);
+        promise.then(loadSchema(type));
+        setState(state => ({ ...state, initialValues: { name, type }, loading: false }));
```

Comment:
In order to pre-populate as many fields as possible, you should just set the response coming back from the API as `initialValues`. It will contain a lot of attributes we don't use, but as we don't have fields for them, it should be safe:
```ruby
API.get(`/api/authentications/${recordId}`).then((initialValues) => {
  ...
  setState(state => ({ ...state, initialValues, loading: false }));

Reply:
After this any field that's not pre-populated upon edit is probably wrongly named, as we had with the `type` vs `credentialType`.